modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,438 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.  
+//
+//
+//                          License Agreement
+//                For Open Source Computer Vision Library
+//
+// Copyright (C) 2020, Huawei Technologies Co., all rights reserved.
+// Third party copyrights are property of their respective owners.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License. 
+//
+// Author: Longbu Wang <riskiest@gmail.com>
+//         Liangqian Kong <chargerKong@126.com>#include <opencv2/core.hpp>
+#include <opencv2/core.hpp>
+#include <iostream>
+namespace cv 
+{
+    /**
+     * Quaternion are a number system that extends the complex numbers.
+     * A quaternion is generally represented in the form:
+     * 		\begin{array}{rcl}
+     * 		q &= w + x\boldsymbol{i} + y\boldsymbol{j} + z\boldsymbol{k}\\
+     *		  &= [w, x, y, z]\\
+     *		  &= [w, \boldsymbol{v}]
+     *		  \end{array}
```

Comment:
You can render Latex formula with Doxygen, [see](https://www.doxygen.nl/manual/formulas.html).

Reply:
Thank you for your code review. I would like to consult where the file generated by doxygen to place?

Reply:
See the corresponding [tutorial](https://docs.opencv.org/master/d4/db1/tutorial_documentation.html).

On Ubuntu, you can install doxygen with something like this:
```
sudo apt-get install doxygen
sudo apt-get install graphviz
sudo apt-get install texlive-latex-extra
```

By the way, to submit a pull request in OpenCV it is better to create a new branch. This way, the PR will be automatically tested on https://pullrequest.opencv.org/#/summary/opencv

Reply:
I'm sorry I may misunderstand what you meant until I see the doxygen warnings. Do you mean that I have to change the comment from latex format to the doxygen format and download the doxygen to check whether my comments are all correct?

Thanks, I'll submit the new pull request after I've modified my code.

Reply:
You can wrap Latex formula with the following Doxygen `\f$` or `\f[` tags to render Latex formula in the generated html docs.
See for instance how it is done [here](https://github.com/opencv/opencv/blob/f52a2cf5e16c5545d8e6cc7bcb1571e4d3c2cddf/modules/calib3d/include/opencv2/calib3d.hpp#L64) and how the documentation looks like [here](https://docs.opencv.org/4.4.0/d9/d0c/group__calib3d.html#details).

For now, the generated documentation looks like [this](https://pullrequest.opencv.org/buildbot/export/pr/18335/docs/da/d4a/classcv_1_1Quat.html#details).

---

I think it is better to ask the core OpenCV dev (**/cc** @alalek ) if it is better to submit a new PR or keep the development in this one.

Reply:
Thanks very much for your help on doxygen


--------------------------------------------------------------------------------------------------------
maybe I could submit the code here first to the subsequent discussion.

Reply:
Thanks again for your help on doxygen
In this commit, we only fix the warning of the doxygen and first put out the code . we will improve the comment later.